includes/Modules/Idea_Hub.php

```diff
@@ -128,6 +159,96 @@ function( $maybe_empty, $postarr ) {
 		$this->post_topic_setting->register();
 	}
 
+	/**
+	 * Shows admin notification for idea hub ideas on post list screen.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $notices Array of admin notices.
+	 *
+	 * @return array Array of admin notices.
+	 */
+	private function googlesitekit_admin_notices( $notices ) {
+		if ( 'edit-post' !== get_current_screen()->id ) {
+			return $notices;
+		}
+
+		$notice_settings = array(
+			'saved' => array(
+				'text' => esc_html__( 'Need some inspiration? Revisit your saved ideas in Site Kit', 'google-site-kit' ),
+				'cta'  => esc_html__( 'See saved ideas', 'google-site-kit' ),
+				'link' => admin_url( 'admin.php?page=googlesitekit-dashboard#new-ideas' ),
+			),
+			'new'   => array(
+				'text' => esc_html__( 'Need some inspiration? Here are some new ideas from Site Kit’s Idea Hub', 'google-site-kit' ),
+				'cta'  => esc_html__( 'See new ideas', 'google-site-kit' ),
+				'link' => admin_url( 'admin.php?page=googlesitekit-dashboard#saved-ideas' ),
+			),
+		);
+
+		$dismissed_items = $this->dismissed_items->get_dismissed_items();
+		$ideas           = $this->transients->get( self::TRANSIENT_IDEAS );
+
+		if ( false === $ideas ) {
+			$ideas = array(
+				'saved-ideas' => $this->get_data( 'saved-ideas' ),
+				'new-ideas'   => $this->get_data( 'new-ideas' ),
+			);
+
+			$this->transients->set( self::TRANSIENT_IDEAS, $ideas, DAY_IN_SECONDS );
+		}
+		$dismissed_items = array();
+
+		foreach ( $ideas as $idea_type => &$all_ideas ) {
+			foreach ( $all_ideas as $k => $idea ) {
+				if ( in_array( $idea['name'], $dismissed_items, true ) ) {
+					unset( $all_ideas[ $k ] );
+				}
+			}
+		}
```

Comment:
I am not sure that I can follow the logic here. How come a saved or a new idea can be added to dismissed items list?

Reply:
In the IB:

```
The notice dismissals for the two scenarios should be separately stored, both on the server-side:
The dismissal for the "New ideas" notice should be stored for 1 week. In other words, if the user after one week still has no saved ideas, the notice should re-appear.
The dismissal for the "Saved ideas" notice should be permanent, in general. However, there is one particular circumstance when it should be cleared: If the user makes a change that results in them to no longer have any saved ideas (for example unsaving the only saved idea or creating a draft post from the only saved idea), the dismissal should be cleared. In other words, once the user later saves something again, the notice should re-appear.
```

So in this case we are dismissing the notice itself and preventing it from being shown on a subsequent load.

Reply:
Yes, I saw it in the IB/AC. The thing that confuses me is why do we need to check the following:

```php
$dismissed_items = $this->dismissed_items->get_dismissed_items();

...

if ( in_array( $idea['name'], $dismissed_items, true ) ) {
```

We don't have anything in the codebase that lets you as a user dismiss a particular idea using our custom dismissed items functionality. An idea may be dismissed, but the information about it is stored on the Idea Hub service, not in the WordPress database.

Reply:
Perhaps I misunderstood the IB, I thought individual ideas could be dismissed too, so that we need need to make sure that there were ideas that were not dismissed prior to rendering the notice.

I based that on the following, but on re-read it talks only about whether the notices are dismissed not each idea.

```
Check if there are saved ideas by using the get_data method of the class with saved-ideas as parameter and whether the notice has not been dismissed by checking if the slug for the saved idea notice does not exist in the list of get_dismissed_items.
```

Reply:
https://github.com/google/site-kit-wp/pull/3668/commits/4e43c593495304cdce3ca1c8f2f3482608bf91d4